src/app/chart-index/chart-index.component.ts

```diff
@@ -28,6 +30,23 @@ export class ChartIndexComponent implements OnInit {
       this.loading = false;
       this.charts = charts;
       this.totalChartsNumber = charts.length;
+    }, () => {
+      this.loading = false;
+      this.apiNotReady = true;
+      IntervalObservable.create(2000)
+        .takeWhile(() => this.apiNotReady)
+        .subscribe(() => {
+          this.chartsService.getCharts().subscribe(charts => {
+            this.apiNotReady = false;
```

Comment:
This code is duplicated. We can extract this anonymous function into a class method.